Remove some inline arguments and make some njit functions global

aseyboldt opened the pull request:
This allows numba to reuse previous typing and compilation
results if the same function is reused, which then also
leads to smaller llvm modules.

For the tests to continue to work we have to return
those global functions through a wrapper (`basic.global_numba_func`)
so that the tests are still able to disable compilation.

Also remove some inline="always" arguments that don't seem
to be helpful.

michaelosthege commented:
@aseyboldt if you rebase the CI should pass

There's one TODO comment in your changes. Do you want to take care of that?
The rest looks fine to me 👍

aseyboldt commented:
> There's one TODO comment in your changes. Do you want to take care of that?

I removed the TODO, I didn't mean to put it in the repo...

Rebased, hopefully it passes now...
